### PR TITLE
Fix SQL direct periodics (replica, encryptionkey, postgres, mysql)

### PR DESCRIPTION
**Pull request by jasonvigil**
See commit message for details

### CODE REVIEW

**Comment by yuwenma** [comment]
/lgtm
/approve

Make sense

File: pkg/controller/direct/sql/sqlinstance_controller.go [hunk 1]
```diff
@@ -347,6 +347,9 @@ func (a *sqlInstanceAdapter) Delete(ctx context.Context, deleteOp *directbase.De
 	if err != nil {
 		return false, fmt.Errorf("deleting SQLInstance %s failed: %w", a.resourceID, err)
 	}
+	if err := a.pollForLROCompletion(ctx, op, "delete"); err != nil {
```
Comment (justinsb) [inline pkg/controller/direct/sql/sqlinstance_controller.go:350]: I would have expected us to pass `deleteOp` into `pollForLROCompletion`

**Review by justinsb — APPROVED** [review]

File: pkg/controller/direct/sql/sqlinstance_controller.go (hunk 1 above)
Comment (jasonvigil) [reply pkg/controller/direct/sql/sqlinstance_controller.go:350]: We need a handle to the LRO to poll / wait. `deleteOp` does not provide that (currently). Probably some way to improve it in the future, but that would likely be a separate effort. We'd need to decide how to make the interface work for services that do not use LROs, and for each service's separate LRO implementation.

Comment (yuwenma) [reply pkg/controller/direct/sql/sqlinstance_controller.go:350]: Thanks. Please update this once you have the follow up LRO improvements

**Pull request merged**